activerecord/test/cases/relation/where_test.rb

```diff
@@ -48,6 +48,13 @@ def test_where_copies_arel_bind_params
       assert_equal [chef], chefs.to_a
     end
 
+    def test_where_with_casted_value_is_null
+      sql = Topic.where(last_read: "").to_sql
+
+      assert_no_match %r{["`]last_read["`] = NULL}, sql
+      assert_match %r{["`]last_read["`] IS NULL}, sql
```

Comment:
Let's use `assert_equal 1, Task.where(starting: "").count` instead. Trying to match quoted identifiers seems tricky, because we'd have to account for SQL Server, etc. (And just in general, if we can prove what we need without being specific about the SQL, so much the better.)